state/state.go

```diff
@@ -1387,3 +1398,101 @@ func escape(value string) string {
 	intermediate = strings.Replace(intermediate, "}", "\\}", -1)
 	return strings.Replace(intermediate, ",", "\\,", -1)
 }
+
+//UnmarshalYAML will unmarshal the helmfile yaml section and fill the SubHelmfileSpec structure
+//this is required to keep allowing string scalar for defining helmfile (maybe)
+func (hf *SubHelmfileSpec) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	var tmp interface{}
+	if err := unmarshal(&tmp); err != nil {
+		return err
+	}
+
+	switch i := tmp.(type) {
+	case string: // single path definition without sub items
+		hf.Path = i
+	case map[interface{}]interface{}: // helmfile path with sub section
+		for k, v := range i {
+			switch key := k.(type) {
+			case string:
+				//get the path
+				if key == "path" {
+					hf.Path = v.(string)
+				} else if key == "selectors" {
+					if err := extractSelectorContent(hf, v); err != nil {
+						return err
+					}
+				} else {
```

Comment:
Can you by any chance remove this implementation of "dict key as the path to a subhelmfile" in favor of `path`, so that the implementation becomes simpler, and there's fewer ways to achieve the same goal?

Reply:
yes, I'll do